likeminds-feed-android-core/src/main/java/com/likeminds/feed/android/core/post/detail/viewmodel/LMFeedPostDetailViewModel.kt

```diff
@@ -0,0 +1,480 @@
+        const val PAGE_SIZE = 10
+        const val REPLIES_PAGE_SIZE = 5
+    }
+
+    // to getPost and paginated comments
+    fun getPost(postId: String, page: Int) {
+        viewModelScope.launchIO {
+            // builds api request
+            val request = GetPostRequest.Builder()
+                .postId(postId)
+                .page(page)
+                .pageSize(PAGE_SIZE)
+                .build()
+
+            // calls api
+            val response = lmFeedClient.getPost(request)
+
+            if (response.success) {
+                val data = response.data ?: return@launchIO
+                val post = data.post
+                val users = data.users
+                val topics = data.topics
+                _postResponse.postValue(
+                    Pair(
+                        page,
+                        LMFeedViewDataConvertor.convertPost(
+                            post,
+                            users,
+                            topics
+                        )
+                    )
+                )
+            } else {
+                errorMessageChannel.send(ErrorMessageEvent.GetPost(response.errorMessage))
+            }
+        }
+    }
+
+    //for like/unlike a comment
+    fun likeComment(
+        postId: String,
+        commentId: String,
+        commentLiked: Boolean
+    ) {
+        viewModelScope.launchIO {
+            val request = LikeCommentRequest.Builder()
+                .postId(postId)
+                .commentId(commentId)
+                .build()
+
+
+            //call like post api
+            val response = lmFeedClient.likeComment(request)
+
+            //check for error
+            if (response.success) {
+                //todo: ask if we have to give callbacks on api success or send events here only
+//                sendCommentLikedEvent(postId, commentId, commentLiked)
+            } else {
+                errorMessageChannel.send(
+                    ErrorMessageEvent.LikeComment(
+                        commentId,
+                        response.errorMessage
+                    )
+                )
+            }
+        }
+    }
+
+    //for adding comment on post
+    fun addComment(
+        postId: String,
+        tempId: String,
+        text: String
+    ) {
+        viewModelScope.launchIO {
+            // initializes temp id for local handling of comment
+
+            // builds api request
+            val request = AddCommentRequest.Builder()
+                .postId(postId)
+                .text(text)
+                .tempId(tempId)
+                .build()
+
+            // calls api
+            val response = lmFeedClient.addComment(request)
+            if (response.success) {
+                val data = response.data ?: return@launchIO
+                val comment = data.comment
+                val users = data.users
+                //todo: ask if we have to give callbacks on api success or send events here only
```

Comment:
please confirm whether we have added this or not.

Reply:
Updated in further PRs